src/styles/features/view/animations/loadingSpin.css

```diff
@@ -0,0 +1,9 @@
+@keyframes loading-spin {
```

Comment:
Is this loading spinner specifically for the .png download that react-loader didn't work with?

Can you provide more detail about how react-loader doesn't work with the .png download?

Reply:
The loading spinner CSS I left in by mistake before I realized react-loader would work fine.

As for the .png download, unlike the other formats, comes straight from cytoscape.js, and is then saved with saveAs from file-saver. I couldn't figure out how to tell when the procedure is done since neither saveAs nor cy.png() return promises or take callbacks.

Reply:
Have you tried something like this already?
```js
this.setState({loading: true});
cy.png(...);
saveAs(blob, ..., ...);
this.setState({loading: false});
```

Reply:
You can also do:

```js
const saveImage = new Promise((resolve, reject) => {
   const img = cy.png();
   saveAs(img, ...., ...);
   resolve(null);
});

saveImage.then(() => this.setState({loading: false});

```

Reply:
It might be that saveAs the problem

Reply:
I'll give that a shot tomorrow with some pathways.

Reply:
That code runs into the same problems as the previous code. I've done more testing on it and it seems to be a synchronous process, that for some reason holds the entire page from rendering while executing. If we do
```js
this.setState({loading: true});
cy.png(...);
saveAs(blob, ..., ...);
this.setState({loading: false});
```
then the loader is correctly set to loading before the process executes, and stops loading afterwards, but never displays while it's active.

@hmtinc suggested wrapping the process in a `setTimeout`. This works as long as it's called as a callback to the first `setState`, i.e.
```js
this.setState({ loading: true }, () => {
  setTimeout(() => {
    saveAs(this.props.cy.png({
      ...
    }), ...);
    this.setState({ loading: false });
  }, 1);
});
```
Doesn't matter how long the timeout is (1ms works fine), and it has to be wrapped in the callback for `this.setState`.